custom-typings/signal-exit/index.d.ts

```diff
@@ -0,0 +1,13 @@
+declare module "signal-exit" {
```

Comment:
From  our discussion in the channel, I think it's better off dropping off TS from this PR, for now.

Reply:
Rewrite this in JS (even not ES6 for now), for backward compatibility.